purchase_sale_inter_company/models/purchase_order.py

```diff
@@ -3,26 +3,76 @@
 # Copyright 2018-2019 Tecnativa - Carlos Dauden
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
 
-from odoo import _, models
+from odoo import _, fields, models
 from odoo.exceptions import UserError
 
 
 class PurchaseOrder(models.Model):
     _inherit = "purchase.order"
 
+    auto_sale_order_info = fields.Html(compute="_compute_auto_sale_order_info")
+
+    def _compute_auto_sale_order_info(self):
+        for purchase_order in self.sudo():
+            auto_sale_order_id = (
+                self.env["sale.order"]
+                .sudo()
+                .search([("auto_purchase_order_id", "=", purchase_order.id)])
+            )
```

Comment:
This `search()` may return a `sale.order` recordset, but at line 28 we have `auto_sale_order_id.display_name` which will raise an error if there's more than 1 SO. Please fix this.